pkg/kubernetes/pods.go

```diff
@@ -9,6 +9,14 @@ import (
 	k8s "k8s.io/client-go/kubernetes"
 )
 
+func ListPodsInterface(client k8s.Interface, labelSelector map[string]string) (*core_v1.PodList, error) {
```

Comment:
Can you please check if *k8s.ClientSet implements k8s.Interface? I remember that it does. In that case you can just modify the other method to take k8s.Interface.

Reply:
Yes it does. I noted that point and I think it should be a separate task as it increases the scope of this PR a lot more than fixing status as ListPods is used in multiple places.

Reply:
If *k8s.ClientSet implements k8s.Interface and you changed ListPod to take k8s.Interface then all existing calls are still valid.